### PR TITLE
redo(ticdc): fix resolved moves too fast when part of tables are not maintained redo writer

### LINKED ISSUE #5486: Redo log: data was lost or damaged in some test cases, and sometimes changefeed failed: "redo log flush fail"
### What did you do?

- Create a changefeed with redo enabled, config:   
> [consistent]
level = "eventual"
storage = "s3://nfs/test-infra-redolog/redo-apply-cdc-all-node-restart-sync?access-key=minioadmin&secret-access-key=minioadmin&endpoint=http://minio.pingcap.net:9000&force-path-style=true"
max-log-size = 1

- Run sysbench :  sysbench oltp_insert prepare --tables=10 --table-size=500 --threads=10 && sysbench oltp_insert  run --tables=10 --table-size=500 --threads=10

- Run  upstream cluster chaos step by step: 
> All PD Restart 
> All capture restart
> All TiKV restart

### What did you expect to see?

- data keeps exactly the same between upstream and downstream, also changefeed status is normal eventually.

_No response_

### What did you see instead?

- data was lost or damaged in test case:
>  redo_apply_cdc_allnode_restart_sync  （failure ratio > 50% ）

> redo_apply_cdc_scale_sync (failure in v6.1.0-pre, http://rms.pingcap.net:31714/artifacts/testground/plan-exec-810836/plan-exec-810836-3745080740/main-logs)

- Somethimes changefeed failed, case log: http://rms.pingcap.net:31714/artifacts/testground/plan-exec-840037/plan-exec-840037-493892576/main-logs

-   {

    "id": "redo-apply-cdc-all-node-restart-sync",
    "summary": {
      "state": "failed",
      "tso": 433315158137241606,
      "checkpoint": "2022-05-19 13:15:48.900",
      "error": {
        "addr": "upstream-ticdc-1.upstream-ticdc-peer.cdc-testbed-tps-840037-1-931.svc:8301",
        "code": "CDC:ErrProcessorUnknown",
        "message": "[CDC:ErrS3StorageAPI]s3 storage api: RequestCanceled: request context canceled\ncaused by: context deadline exceeded"
      }
 
cdc.log ERROR:
  [2022/05/19 13:20:33.310 +00:00] [ERROR] [file.go:199] ["redo log flush fail"] [namespace=default] [changefeed=redo-apply-cdc-all-node-restart-sync] [error="[CDC:ErrS3StorageAPI]s3 storage api: RequestCanceled: request context canceled\ncaused by: context deadline exceeded"]

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
TiKV
Release Version:   6.1.0-alpha
Edition:           Community
Git Commit Hash:   2accd27c2a3882d3d45b6958565efb6de4d6e33c
Git Commit Branch: heads/refs/tags/v6.1.0-nightly
UTC Build Time:    2022-05-19 11:14:52
Rust Version:      rustc 1.60.0-nightly (1e12aef3f 2022-02-13)
Enable Features:   jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
Release Version: v6.1.0-nightly
Git Commit Hash: 5e07c8b6bbefa7b087933ab2c12358cf5c05d76d
Git Branch: heads/refs/tags/v6.1.0-nightly
UTC Build Time: 2022-05-19 11:12:29
Go Version: go version go1.18.2 linux/amd64
Failpoint Build: false
```

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #5486

### What is changed and how it works?

redo manager queries redo log writer to update table resolved ts, but when part of tables are not maintained in redo log writer, the redo manager doesn't respect the resolved ts of these tables and could move resolved ts forward too fast, which leads to data loss.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
Fix a bug that resolved ts moves too fast when part of tables are not maintained redo writer.
```

### CODE REVIEW

File: cdc/redo/manager.go [hunk 1]
```diff
@@ -360,8 +360,11 @@ func (m *ManagerImpl) updateTableResolvedTs(ctx context.Context) error {
 		return err
 	}
 	minResolvedTs := uint64(math.MaxUint64)
-	for tableID, rts := range rtsMap {
-		m.rtsMap[tableID] = rts
+	for tableID := range m.rtsMap {
+		if rts, ok := rtsMap[tableID]; ok {
```
Comment (nongfushanquan) [inline cdc/redo/manager.go:364]: why part of tables are not maintained in redo log writer?

Comment (amyangfei) [reply cdc/redo/manager.go:364]: If there is no data of this table, and flush resolved ts of this table has not been executed, the redo log writer doesn't know this table.
The redo log writer only records the information of the table that it has met.

**Review by nongfushanquan — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**